internal/resources/app_endpoint_activation_status.go

```diff
@@ -0,0 +1,368 @@
+// Couchbase Capella's v4 does not support a GET endpoint for activation state directly.
+// This read is calling the retrieveAppEndpointActivation func to verify the state with the app endpoint response.
+// API reference: Post/Del activation status and Get App Endpoint
+// - POST: App Endpoint online
+// - DELETE: App Endpoint offline
+// - GET: App Endpoint details (contains state)
+// See: https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/App-Endpoints/operation/postAppEndpointActivationStatus
+// See: https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/App-Endpoints/operation/deleteAppEndpointActivationStatus
+// See: https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/App-Endpoints/operation/getAppEndpoint
+func (r *AppEndpointActivationStatus) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.AppEndpointActivationStatus
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading app endpoint activation details in Capella",
+			"Could not validate the app endpoint activation for app endpoint "+state.AppEndpointName.String()+": "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId  = IDs[providerschema.OrganizationId]
+		projectId       = IDs[providerschema.ProjectId]
+		clusterId       = IDs[providerschema.ClusterId]
+		appServiceId    = IDs[providerschema.AppServiceId]
+		appEndpointName = IDs[providerschema.AppEndpointName]
+	)
+
+	refreshedState, err := r.waitForAppEndpointStatus(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, state.Online.ValueBool())
+	if err != nil {
+		resourceNotFound, _ := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error parsing read app endpoint activation request",
+			"Could not read the app endpoint details, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, &refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update switches the app endpoint to online/offline state.
+func (r *AppEndpointActivationStatus) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan providerschema.AppEndpointActivationStatus
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := r.validate(plan); err != nil {
+		resp.Diagnostics.AddError(
+			"Error parsing update app endpoint activation status request",
+			"Could not switch app endpoint online/offline, unexpected error: "+err.Error(),
+		)
+		return
```

Comment:
validation should be done at schema level. see this for an example:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/main/internal/resources/free_tier_cluster_on_off_schema.go#L9